jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/annotations/PreDestoryAnnotationQuickFix.java

```diff
@@ -0,0 +1,101 @@
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.IBinding;
+import org.eclipse.jdt.core.dom.IMethodBinding;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.SingleVariableDeclaration;
+import org.eclipse.jdt.core.dom.VariableDeclarationFragment;
+import org.eclipse.jdt.internal.corext.dom.Bindings;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4jakarta.jdt.codeAction.IJavaCodeActionParticipant;
+import org.eclipse.lsp4jakarta.jdt.codeAction.JavaCodeActionContext;
+import org.eclipse.lsp4jakarta.jdt.codeAction.proposal.ChangeCorrectionProposal;
+import org.eclipse.lsp4jakarta.jdt.codeAction.proposal.ModifyModifiersProposal;
+import org.eclipse.lsp4jakarta.jdt.codeAction.proposal.RemoveParamsProposal;
+import org.eclipse.lsp4jakarta.jdt.core.annotations.AnnotationConstants;
+import org.eclipse.lsp4jakarta.jdt.core.beanvalidation.BeanValidationConstants;
+
+
+/**
+ * Quickfix for annotation PreDestory 
+ * 1. Removing static from a static method
+ * 2. Removing all parameters from method
+ * 
+ * @author Zijian Pei
+ *
+ */
+public class PreDestoryAnnotationQuickFix implements IJavaCodeActionParticipant {
+	
+	public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
+            IProgressMonitor monitor) throws CoreException {
+        ASTNode node = context.getCoveredNode();
+        MethodDeclaration parentNode = (MethodDeclaration) node.getParent();
+        IBinding parentMethod = getBinding(node);
+        
+        MethodDeclaration parentNode2 = (MethodDeclaration) node.getParent();
+        IMethodBinding parentMethod2 = parentNode2.resolveBinding();
+
+        List<CodeAction> codeActions = new ArrayList<>();
+        List<SingleVariableDeclaration> parameters = (List<SingleVariableDeclaration>) parentNode.parameters();
+
+
+        if (diagnostic.getCode().getLeft().equals(AnnotationConstants.DIAGNOSTIC_CODE_PREDESTROY_STATIC)) {
+            codeActions.add(ChangeToNonStatic(diagnostic, context, parentMethod));
+        }
+        
+        if (diagnostic.getCode().getLeft().equals(AnnotationConstants.DIAGNOSTIC_CODE_PREDESTROY_PARAMS)) {           
+                String name = "Remove all parameters";
+                ChangeCorrectionProposal proposal = new RemoveParamsProposal(name,
+                        context.getCompilationUnit(), context.getASTRoot(), parentMethod2, 0, parameters, null);
+                // Convert the proposal to LSP4J CodeAction
+                CodeAction codeAction = context.convertToCodeAction(proposal, diagnostic);
+                codeActions.add(codeAction);  
+        }
+        
+
+        return codeActions;
+    }
+	
+	
+	private CodeAction ChangeToNonStatic(Diagnostic diagnostic, JavaCodeActionContext context, IBinding parentType) throws CoreException {
```

Comment:
follow proper naming conventions, name should be "changeToNonStatic"

Reply:
I create a separate class to solve this